igmspec/ingest/myers.py

```diff
@@ -4,13 +4,143 @@
+    # Myers master QSO catalog
+    ADM_file = os.getenv('RAW_IGMSPEC') + '/Myers/GTR-ADM-QSO-master-wvcv.fits'
+    ADM_qso = Table.read(ADM_file)
+    head1 = fits.open(ADM_file)[1].header
+    DATE = head1['DATE']
+
+    # Photometry for Myers QSO catalog. This file is not aligned with the catalog file, i.e. it is a
+    # superset that includes the catalog file. For that reason we need to match and tack on photometry
+    ADM_sweep_file = os.getenv('RAW_IGMSPEC') + '/Myers/GTR-ADM-QSO-master-sweeps-Feb5-2016.fits'
+    ADM_sweep = Table.read(ADM_sweep_file)
+
+    c_qso = SkyCoord(ra=ADM_qso['RA'] * u.deg, dec=ADM_qso['DEC'] * u.deg)
+    c_swp = SkyCoord(ra=ADM_sweep['RA'] * u.deg, dec=ADM_sweep['DEC'] * u.deg)
+
+    ## Create an aligned Table for matching photometry from sweeps
+    nqso = len(ADM_qso)
+    qso_phot = Table(np.repeat(np.zeros_like(ADM_sweep[0]), nqso))
+    # Rename the RA and DEC
+    qso_phot.rename_column('RA', 'RA_sweep')
+    qso_phot.rename_column('DEC', 'DEC_sweep')
+    # Cull out the keys which already exist in the ADM_qso Table (except
+    # for the RA and DEC, which we renamed)
+    dupe_keys = list(set(ADM_qso.keys()) & set(qso_phot.keys()))
+    qso_phot.remove_columns(dupe_keys)
+
+    idx, d2d, d3d = c_qso.match_to_catalog_sky(c_swp)
+    # Currently using 1.0" for matching, as for the SDSS objects, these will mostly be the exact
+    # same coordinates.
+    itrim = (d2d <= 1.0 * u.arcsec)
+    qso_phot[:][itrim] = ADM_sweep[:][idx[itrim]]
+    ADM_qso = hstack([ADM_qso, qso_phot], join_type='exact')
+    # Trim to only spectroscopic objects
+    ispec = spectro_myers(ADM_qso)
+    ADM_qso = ADM_qso[ispec]
+    # assign best redshifts to ZEM tag
+    zbest_myers(ADM_qso)
+    # Add MYERS prefix to all MYERS tags
+    for key in ADM_qso.keys():
+        ADM_qso.rename_column(key, 'MYERS_' + key)
+
+    # Now we meatch the SDSS/BOSS and Myers catalogs to create one master QSO catalog
+    #
+    # There are three groups of objects, 1) SDSS-MYERS match, 2) SDSS only, 3) Myers only.
+    # Deal with each in turn.
+
+    # 1) SDSS-MYERS match. Add Myers tags to the SDSS structure
+    c_sdss = SkyCoord(ra=sdss_boss1['SDSS_PLUG_RA'] * u.deg, dec=sdss_boss1['SDSS_PLUG_DEC'] * u.deg)
+    c_myers = SkyCoord(ra=ADM_qso['MYERS_RA'] * u.deg, dec=ADM_qso['MYERS_DEC'] * u.deg)
+    isdss, imyers, d2d, _ = search_around_sky(c_sdss, c_myers, MATCH_TOL)
+    sdss_myers = hstack([sdss_boss1[isdss], ADM_qso[imyers]], join_type='exact')
+    sdss_myers['SDSS_MYERS_FLAG'] = 'SDSS_MYERS'
+    sdss_myers['RA'] = sdss_myers['SDSS_PLUG_RA']
+    sdss_myers['DEC'] = sdss_myers['SDSS_PLUG_DEC']
+    sdss_myers['SOURCEBIT'] = sdss_myers['MYERS_SOURCEBIT']
+    sdss_myers['ZEM'] = sdss_myers['MYERS_ZEM']
+    sdss_myers['ZEM_SOURCE'] = sdss_myers['MYERS_ZEM_SOURCE']
+
+    # 2) SDSS only
+    # Find the SDSS objects that have no match in the Myers catalog
+    inomatch = np.ones(len(c_sdss), dtype=bool)
```

Comment:
Changed back. Code won't run with ones_like. This is also bad from a memory perspective.